pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -126,7 +126,7 @@ def setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", stage: O
         if not trainer.logger:
             raise MisconfigurationException("Cannot use GPUStatsMonitor callback with Trainer that has no logger.")
 
-        if trainer._device_type != _AcceleratorType.GPU:
+        if trainer._device_type != _AcceleratorType.GPU.lower():
```

Comment:
The change is not required?

Reply:
Yeah, it's wired. Most other places "cpu" == "CPU" works, but not here, if I don't convert to lower, this check failed. 
I saw lite cast to lower too, any idea why sometimes lower() is needed sometimes doesn't? https://github.com/PyTorchLightning/pytorch-lightning/blob/0adcd6a048b66eed8724c49e7ff4e641044e82db/pytorch_lightning/lite/lite.py#L446
cc: @awaelchli

Reply:
It shouldn't when checking with the enum because we already do it:

https://github.com/PyTorchLightning/pytorch-lightning/blob/43a89eb132511f99441671ee0ba4d2b92ebbcbd7/pytorch_lightning/utilities/enums.py#L36-L38

Reply:
It works everywhere, but not here. I have no idea why